src/js/actions/search/layers.js

```diff
@@ -87,55 +121,68 @@ define(function (require, exports) {
             document = appStore.getCurrentDocument();
-        if (!document) {
-            return {};
-        }
-        var layers = document.layers.allVisibleReversed,
-            layerMap = layers.map(function (layer) {
-                var ancestry = _formatLayerAncestry(layer, appStore),
-                    layerType = _getLayerCategory(layer),
-                    iconID = svgUtil.getSVGClassFromLayer(layer);
-
-                return {
-                    id: layer.id.toString(),
-                    name: layer.name,
-                    pathInfo: ancestry,
-                    iconID: iconID,
-                    category: layerType
-                };
-            }.bind(this));
 
-        return layerMap;
+        return _getLayerOptionsFromDoc.call(this, document, "CURRENT");
     };
 
     /**
-     * Select layer when its item is confirmed in search
+     * Make list of certain layer info from all documents so search store can create search options
      *
      * @private
-     * @param {number} idInt ID of layer to select
+     * @return {Immutable.List.<object>}
     */
-    var _confirmSearch = function (idInt) {
+    var _getAllLayerSearchOptions = function () {
+        // Get list of layers
         var appStore = this.flux.store("application"),
-            document = appStore.getCurrentDocument(),
-            selected = document.layers.byID(idInt);
+            currDoc = appStore.getCurrentDocument(),
+            documents = appStore.getOpenDocuments(),
+            // add layers from current document first, then skip it in the loop
+            allLayerMaps = _getLayerOptionsFromDoc.call(this, currDoc, "ALL");
+
+        documents.forEach(function (document) {
+            if (document !== currDoc) {
+                var layerMap = _getLayerOptionsFromDoc.call(this, document, "ALL");
+                allLayerMaps = allLayerMaps.concat(layerMap);
+            }
+        }.bind(this));
+
+        return allLayerMaps;
+    };
+
+    /**
+     * Select layer when its item is confirmed in search
+     *
+     * @private
+     * @param {string} id ID of layer to select
+    */
+    var _confirmSearch = function (id) {
+        var docStore = this.flux.store("document"),
+            splitID = id.split("_"),
+            docID = mathUtil.parseNumber(splitID[0]),
+            layerID = mathUtil.parseNumber(splitID[1]),
+            document = docStore.getDocument(docID),
+            selected = document.layers.byID(layerID);
 
         if (selected) {
+            this.flux.actions.documents.selectDocument(document);
             this.flux.actions.layers.select(document, selected);
         }
     };
 
     /**
-     * Register layer info for search
+     * Register layer info to search
      *
+     * @param {string} type "CURRENT" or "ALL". If not specified, defaults to current
      */
-    var registerLayerSearch = function () {
+    var _registerLayerSearch = function (type) {
```

Comment:
It would be nicer to make this a boolean parameter so that the detail of which strings in particular are valid can be more limited.